src/intersections/segments.jl

```diff
@@ -167,3 +167,268 @@ function intersectcolinear(s1::Segment{Dim,T}, s2::Segment{Dim,T}) where {Dim,T}
     NoIntersection()
   end
 end
+
+"""
+    intersecttype(s, t)
+
+Calculate the intersection of a segment and triangle in 3D.
+Algorithm from Jiménez et al 2009 (A robust segment/triangle intersection algorithm for interference tests. Efficiency study). 
+"""
+function intersecttype(s::Segment{3,T}, t::Triangle{3,T}) where {T}
```

Comment:
Let's add this implementation to a separate source file about triangles, `triangles.jl`